react/src/utils/downloadCsv.tsx

```diff
@@ -16,13 +17,16 @@ const objectToCsv = <T extends object>(data: T[]) => {
     return csvRows.join('\n');
 };
 
-const download = (data: BlobPart) => {
+const formatFileName = (filename: string) =>
+    filename.endsWith('.csv') ? filename : `${filename}.csv`;
+
+const download = (data: BlobPart, filename: string) => {
     const blob = new Blob([data], { type: 'text/csv' });
     const url = window.URL.createObjectURL(blob);
     const a = document.createElement('a');
     a.setAttribute('hidden', '');
     a.setAttribute('href', url);
-    a.setAttribute('download', 'ssmp.csv');
+    a.setAttribute('download', formatFileName(filename));
     document.body.appendChild(a);
     a.click();
     document.body.removeChild(a);
```

Comment:
I know this code was already here rather than modified by the PR, but in Stager we don't even have to append the element to the body or set the a as hidden. The blob URL also isn't freed after the click.